src/uucore/src/lib/features/selinux.rs

```diff
@@ -387,19 +483,146 @@ mod tests {
+    fn test_contexts_differ() {
+        let file1 = NamedTempFile::new().expect("Failed to create first tempfile");
+        let file2 = NamedTempFile::new().expect("Failed to create second tempfile");
+        let path1 = file1.path();
+        let path2 = file2.path();
+
+        std::fs::write(path1, b"content for file 1").expect("Failed to write to first tempfile");
+        std::fs::write(path2, b"content for file 2").expect("Failed to write to second tempfile");
+
+        if !is_selinux_enabled() {
+            assert!(
+                contexts_differ(path1, path2),
+                "contexts_differ should return true when SELinux is not enabled"
+            );
+            return;
+        }
+
+        let test_context = String::from("system_u:object_r:tmp_t:s0");
+        let result1 = set_selinux_security_context(path1, Some(&test_context));
+        let result2 = set_selinux_security_context(path2, Some(&test_context));
+
+        if result1.is_ok() && result2.is_ok() {
+            assert!(
+                !contexts_differ(path1, path2),
+                "Contexts should not differ when the same context is set on both files"
+            );
+
+            let different_context = String::from("system_u:object_r:user_tmp_t:s0");
+            if set_selinux_security_context(path2, Some(&different_context)).is_ok() {
+                assert!(
+                    contexts_differ(path1, path2),
+                    "Contexts should differ when different contexts are set"
+                );
+            }
+        } else {
+            println!(
+                "Note: Couldn't set SELinux contexts to test differences. This is expected if the test doesn't have sufficient permissions."
+            );
+            assert!(
+                contexts_differ(path1, path2),
+                "Contexts should differ when different contexts are set"
+            );
+        }
+
+        let nonexistent_path = Path::new("/nonexistent/file/path");
+        assert!(
+            contexts_differ(path1, nonexistent_path),
+            "contexts_differ should return true when one path doesn't exist"
+        );
+    }
+
+    #[test]
+    fn test_preserve_security_context() {
+        let source_file = NamedTempFile::new().expect("Failed to create source tempfile");
+        let dest_file = NamedTempFile::new().expect("Failed to create destination tempfile");
+        let source_path = source_file.path();
+        let dest_path = dest_file.path();
+
+        std::fs::write(source_path, b"source content").expect("Failed to write to source tempfile");
+        std::fs::write(dest_path, b"destination content")
+            .expect("Failed to write to destination tempfile");
+
+        if !is_selinux_enabled() {
+            let result = preserve_security_context(source_path, dest_path);
+            assert!(
+                result.is_ok(),
+                "preserve_security_context should succeed when SELinux is not enabled"
+            );
```

Comment:
This assert currently fails because `preserve_security_context` now returns an error if selinux is not enabled.